src/main/scala/shine/cuda/Kernel.scala

```diff
@@ -22,42 +22,38 @@ case class Kernel(decls: Seq[C.AST.Decl],
                  ) extends util.Kernel(decls, kernel, outputParam, inputParams, intermediateParams, printer) {
 
   //TODO: how does this works for CUDA?
-  override protected def findParameterMappings(arguments: List[Argument], localSize: LocalSize, globalSize: GlobalSize): Map[Nat, Nat] = {
-    val numGroups: NDRange = (
-      globalSize.size.x /^ localSize.size.x,
-      globalSize.size.y /^ localSize.size.y,
-      globalSize.size.z /^ localSize.size.z)
+  override protected def findParameterMappings(arguments: List[Argument], localSize: LocalSize, globalSize: GlobalSize, gridSize: GridSize): Map[Nat, Nat] = {
     val sizeVarMapping = collectSizeVars(arguments, Map(
-      get_num_groups(0) -> numGroups.x,
-      get_num_groups(1) -> numGroups.y,
-      get_num_groups(2) -> numGroups.z,
-      get_local_size(0) -> localSize.size.x,
-      get_local_size(1) -> localSize.size.y,
-      get_local_size(2) -> localSize.size.z,
-      get_global_size(0) -> globalSize.size.x,
-      get_global_size(1) -> globalSize.size.y,
-      get_global_size(2) -> globalSize.size.z
+      gridDim('x') -> gridSize.size.x,
+      gridDim('y') -> gridSize.size.y,
+      gridDim('z') -> gridSize.size.z,
+      blockDim('x') -> localSize.size.x,
+      blockDim('y') -> localSize.size.y,
+      blockDim('z') -> localSize.size.z,
+      globalDim('x') -> globalSize.size.x,
+      globalDim('y') -> globalSize.size.y,
+      globalDim('z') -> globalSize.size.z
     ))
```

Comment:
Man könnte es halt so machen.
Dann hat man nur ein bisschen das Problem mit dem unused Parameter

Reply:
ist der aufruf dann immer noch der selbe?
oder muss man dann für cuda und opencl unterschiedlich die dim angeben?

Reply:
Der Aufruf ist immer noch der Selbe.
Es wurde nur die Berechnung der numGroups in die util-Klasse verlagert und unbennant zu GridSize. Der OpenCL-Executor braucht ja auch irgendwie die numGroups, weshalb eine Berechnung davon in der Oberklasse Sinn macht.
Ich bin mir allerdings nicht so sicher, ob die Bennennung zu GridSize so viel Sinn macht, weil das dann auch im OpenCL-Executor verwendet werden sollte...